<launch>
	<arg name="robot_name" default="robot1"/>
	<arg name="robot_name_slash" default="robot1/"/>

    <group ns="$(arg robot_name)">
        <node name="ekf" pkg="eurobot_localization" type="ekf" >
            <rosparam file="$(find eurobot_localization)/config/global_filter/global_filter.yaml"/>
			<param name="robot_name" value="$(arg robot_name_slash)"/>
            <remap from="odom" to="global_filter"/>
        </node>
    </group>


	<include file="$(find lidar_localization)/launch/lidar_localization_2023.launch">
		<arg name="robot_name" default="$(arg robot_name)"/>
		<arg name="robot_name_slash" default="$(arg robot_name_slash)"/>
    </include>

    <!-- <node name="area_obstacles_extractor" pkg="lidar_localization" type="area_obstacles_extractor_node" output="screen">
        <remap from="robot_pose" to="ekf_pose_in_odom"/>

        <param name="avoid_max_distance" value="0.4"/>
        <param name="avoid_min_distance" value="0.17"/>
	    <param name="ally_excluded_radius" value = "0.15"/>

        <param name="x_min_range" value="0.2"/>
        <param name="x_max_range" value="1.8"/>
        <param name="y_min_range" value="0.2"/>
        <param name="y_max_range" value="2.8"/>

        <rosparam param="excluded_x">[0.842, 0.25, 0.3]</rosparam>
        <rosparam param="excluded_y">[0.148, 0.5, 1.5]</rosparam>
        <rosparam param="excluded_radius">[0.3, 0.3, 0.4]</rosparam>
    </node> -->

    <!-- <node name="ally_area_obstacles_extractor" pkg="lidar_localization" type="area_obstacles_extractor_node" output="screen">
        <remap from="obstacles_to_map" to="ally_obstacle_to_map"/>
        <remap from="obstacle_marker" to="ally_obstacle_marker"/>
        <remap from="robot_pose" to="ekf_pose_in_odom"/>

        <param name="avoid_max_distance" value="0.4"/>
        <param name="avoid_min_distance" value="0.2"/>

        <param name="x_min_range" value="0.15"/>
        <param name="x_max_range" value="1.85"/>
        <param name="y_min_range" value="0.15"/>
        <param name="y_max_range" value="2.85"/>

        <param name="x_min_range" value="0.2"/>
        <param name="x_max_range" value="1.8"/>
        <param name="y_min_range" value="0.2"/>
        <param name="y_max_range" value="2.8"/>

        <rosparam param="excluded_x">[0.842, 0.25, 0.3]</rosparam>
        <rosparam param="excluded_y">[0.148, 0.5, 1.5]</rosparam>
        <rosparam param="excluded_radius">[0.3, 0.3, 0.4]</rosparam>
    </node> -->
</launch>
